core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerPool.scala

```diff
@@ -199,10 +200,9 @@ class ContainerPool(childFactory: ActorRefFactory => ActorRef,
             }
             if (!isResentFromBuffer) {
               // Add this request to the buffer, as it is not there yet.
-              runBuffer = runBuffer.enqueue(r)
+              runBuffer = runBuffer.enqueue(Run(r.action, r.msg, retryLogDeadline))
             }
-            // As this request is the first one in the buffer, try again to execute it.
-            self ! Run(r.action, r.msg, retryLogDeadline)
+          //buffered items will be processed via processBufferOrFeed()
```

Comment:
can you confirm my understanding that this change is sound because the request will still be in the process run buffer?

Reply:
I'm seeing a problem where in some cases messages are processed twice. I will work that out and it may affect this. Will get back to you.

Reply:
Sorry thought you were asking something else. Yes, the runBuffer processing works by:
* when an activation does not get scheduled AND it was not already buffered previously, add it to the buffer
* when an activation does not get scheduled AND it was already buffered - do nothing (it remains at buffer head)
* when an activation gets scheduled to a container, if it was previously buffered, remove it from the buffer, and trigger the next buffered item to be resent
* when some container action occurs (NeedsWork/ContainerRemoved) this also triggers next buffered item to be resent - so we need to be careful about sending duplicates by tracking which item has been resent